src/v/storage/segment.h

```diff
@@ -40,27 +40,70 @@ struct segment_closed_exception final : std::exception {
 class segment {
 public:
     using generation_id = named_type<uint64_t, struct segment_gen_tag>;
-    struct offset_tracker {
+    class offset_tracker {
+    public:
+        using committed_offset_t
+          = named_type<model::offset, struct committed_offset_tag>;
+        using dirty_offset_t
+          = named_type<model::offset, struct dirty_offset_tag>;
+        using stable_offset_t
+          = named_type<model::offset, struct stable_offset_tag>;
+
         offset_tracker(model::term_id t, model::offset base)
-          : term(t)
-          , base_offset(base)
-          , committed_offset(model::prev_offset(base))
-          , dirty_offset(model::prev_offset(base))
-          , stable_offset(model::prev_offset(base)) {}
-        model::term_id term;
-        model::offset base_offset;
+          : _term(t)
+          , _base_offset(base)
+          , _committed_offset(model::prev_offset(base))
+          , _stable_offset(model::prev_offset(base))
+          , _dirty_offset(model::prev_offset(base)) {}
+
+        template<typename... Ts>
+        void set_offsets(Ts&&... ts) {
```

Comment:
nit: this looks fine, but I think you can remove all the forwarding references and pass around `T` values. the model::offset types are just a single integer / trivially copyable.

Reply:
Fair! Forwarding references changed to pass by value.